tests/vmi_configuration_test.go

```diff
@@ -2508,15 +2508,27 @@ var _ = Describe("Configurations", func() {
 		vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
 		Expect(err).ToNot(HaveOccurred(), "Should successfully get VMI")
 
-		vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+		vmiVirtLauncherPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
 
 		tests.WaitAgentConnected(virtClient, vmi)
 
-		// Ensure that VMI IP Stays equal to PodIP even after Guest-Agent kicks in
+		// Ensure that VMI 'ipAddress' and 'ipAddresses' stays equal to
+		// VMI virt-launcher pod 'PodIP' and 'PodIPs' even after Guest-Agent kicks in
 		Consistently(func() bool {
 			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred(), "Should successfully get VMI")
-			return vmi.Status.Interfaces[0].IP == vmiPod.Status.PodIP
+
+			isPodIPSynced := vmiVirtLauncherPod.Status.PodIP == vmi.Status.Interfaces[0].IP
+
+			isPodIPsSynced := true
+			for idx := 0; idx < len(vmiVirtLauncherPod.Status.PodIPs); idx++ {
```

Comment:
I think you should assert the length of the IPs between pod / VMI.

I also personally prefer to iterate the IPs rather than use indexes; I'd use  `for _, ip := range vmiVirtLauncherPod.Status.PodIPs { ... }` instead, but up to you.

Reply:
I iterate by an index to check the order of the addresses as well, I mention it in the commit message.